quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::time::Timestamp;
+use core::{cmp::max, time::Duration};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+/// Bandwidth-related data tracked for each sent packet
+pub struct PacketInfo {
+    /// `Estimator.delivered_bytes` at the time this packet was sent.
+    pub delivered_bytes: u64,
+    /// `Estimator.delivered_time` at the time this packet was sent.
+    pub delivered_time: Timestamp,
+    /// `Estimator.lost_bytes` at the time this packet was sent.
+    pub lost_bytes: u64,
+    /// `Estimator.first_sent_time` at the time this packet was sent.
+    pub first_sent_time: Timestamp,
+    /// The volume of data that was estimated to be in flight at the time of the transmission of this packet
+    pub bytes_in_flight: u32,
+    /// Whether the path send rate was limited by the application rather than congestion control at the time
+    /// this packet was sent
+    pub is_app_limited: bool,
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+/// A bandwidth delivery rate estimate with associated metadata
+pub struct RateSample {
+    /// The length of the sampling interval
+    pub interval: Duration,
+    /// The amount of data in bytes marked as delivered over the sampling interval
+    pub delivered_bytes: u64,
+    /// The amount of data in bytes marked as lost over the sampling interval
+    pub lost_bytes: u64,
+    /// `PacketInfo.is_app_limited` from the most recent acknowledged packet
+    pub is_app_limited: bool,
+    /// `PacketInfo.delivered_bytes` from the most recent acknowledged packet
+    pub prior_delivered_bytes: u64,
+    /// `PacketInfo.bytes_in_flight` from the most recent acknowledged packet
+    pub bytes_in_flight: u32,
+    /// `PacketInfo.lost_bytes` from the most recent acknowledged packet
```

Comment:
should these be referring to PacketInfo or Estimator? Also will this struct see more functions added to it in the future?